Source/WebKit/WebProcess/GPU/media/SourceBufferPrivateRemote.cpp

```diff
@@ -185,20 +185,11 @@ void SourceBufferPrivateRemote::setMode(SourceBufferAppendMode mode)
     m_gpuProcessConnection->connection().send(Messages::RemoteSourceBufferProxy::SetMode(mode), m_remoteSourceBufferIdentifier);
 }
 
-void SourceBufferPrivateRemote::updateBufferedFromTrackBuffers(bool sourceIsEnded)
+void SourceBufferPrivateRemote::clientReadyStateChanged(bool sourceIsEnded)
 {
-    if (!m_mediaSourcePrivate)
-        return;
-
-    if (!isGPURunning())
-        return;
-
-    auto sendResult = m_gpuProcessConnection->connection().sendSync(Messages::RemoteSourceBufferProxy::UpdateBufferedFromTrackBuffers(sourceIsEnded), m_remoteSourceBufferIdentifier);
-    if (!sendResult)
-        return;
-
-    auto [buffered] = sendResult.takeReply();
-    setBufferedRanges(WTFMove(buffered));
+    if (isGPURunning())
+        m_gpuProcessConnection->connection().send(Messages::RemoteSourceBufferProxy::ClientReadyStateChanged(sourceIsEnded), m_remoteSourceBufferIdentifier);
+    SourceBufferPrivate::updateBufferedFromTrackBuffers(m_trackBufferRanges, sourceIsEnded);
```

Comment:
This is a slight change of behaviour: do we care to call this if gpu process is not running?

Reply:
probably not, but it would give a better answer than not. though it probably no longer matters at this stage